x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/common_attributes.gen.ts

```diff
@@ -205,3 +321,78 @@ export const RelatedIntegration = z.object({
 
 export type RelatedIntegrationArray = z.infer<typeof RelatedIntegrationArray>;
 export const RelatedIntegrationArray = z.array(RelatedIntegration);
+
+export type InvestigationFields = z.infer<typeof InvestigationFields>;
+export const InvestigationFields = z.object({
+  field_names: z.array(z.string()).min(1),
+});
+
+/**
+ * Defines the interval on which a rule's actions are executed.
+ */
+export type RuleActionThrottle = z.infer<typeof RuleActionThrottle>;
+export const RuleActionThrottle = z.union([
+  z.enum(['no_actions', 'rule']),
+  z.string().regex(/^[1-9]\d*[smhd]$/),
+]);
+
+/**
+ * The action frequency defines when the action runs (for example, only on rule execution or at specific time intervals).
+ */
+export type RuleActionFrequency = z.infer<typeof RuleActionFrequency>;
+export const RuleActionFrequency = z.object({
+  summary: z.boolean(),
```

Comment:
We have comments for many properties and types in the legacy schema. I noticed that many of them are missing in the new schema. We should move them to the new one if they are well-written. For instance, there's a good clarifying comment for the `summary` field:

```ts
/**
 * Action summary indicates whether we will send a summary notification about all the generate alerts or notification per individual alert
 */
export type RuleActionSummary = t.TypeOf<typeof RuleActionSummary>;
export const RuleActionSummary = t.boolean;
```

We can do it in follow-up PRs, of course.